packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +141,133 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                shippingAddress,
+                lineItems,
+            };
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+
+            await this._store.dispatch(this._consignmentActionCreator.updateConsignment(consignment));
+
+            const submitOrderPayload = {};
+            const submitOrderOptions = {
+                params: {
+                    methodId,
+                },
+            };
+
+            await this._store.dispatch(this._orderActionCreator.submitOrder(submitOrderPayload, submitOrderOptions));
+
+            const paymentData =  {
+                formattedPayload: {
+                    vault_payment_instrument: null,
+                    set_as_default_stored_instrument: null,
+                    device_info: null,
+                    method_id: methodId,
+                    paypal_account: {
+                        order_id: data.orderID,
+                    },
+                },
+            };
+            await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId,paymentData}));
+            window.location.assign('/checkout/order-confirmation');
+        } catch (e) {
+            throw new RequestError(e);
+        }
+    }
+
+    private async _onShippingChangeHandler(data: ShippingChangeData) {
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const { id: selectedShippingOptionId } = data.selected_shipping_option || {};
+        const shippingAddress = await this._transformToAddress(data.shipping_address);
+        this._currentShippingAddress = shippingAddress;
+        const lineItems = this._getLineItems();
+        const consignment = { shippingAddress, lineItems };
+        const existingConsignments = state.consignments.getConsignmentsOrThrow();
+        if (existingConsignments[0]) {
+            await this._store.dispatch(this._consignmentActionCreator.updateConsignment({id: existingConsignments[0].id, ...consignment}));
+        } else {
+            await this._store.dispatch(this._consignmentActionCreator.createConsignments([consignment]));
+        }
+        const updatedConsignment = state.consignments.getConsignmentsOrThrow()[0];
+        const { availableShippingOptions } = updatedConsignment;
+        const { id: recommendedShippingOptionId } = availableShippingOptions?.find(option => option.isRecommended) || {};
+        const isSelectedOptionExist = selectedShippingOptionId && availableShippingOptions?.find(option => option.id === selectedShippingOptionId);
+        await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+        const shippingOptionId = isSelectedOptionExist ? selectedShippingOptionId : recommendedShippingOptionId;
+
+        await this._store.dispatch(this._consignmentActionCreator.selectShippingOption(
+            shippingOptionId || ''
+        ));
```

Comment:
Should we call `selectShippingOption` if `shippingOptionId` is not defined?

Reply:
fixed